10-gis.Rmd

```diff
@@ -895,6 +897,70 @@ But the same is true for the lightweight SQLite/SpatiaLite database engine and G
+
+The SpatioTemporal Asset Catalog (STAC) is a general description format for spatiotemporal data that is used to describe a variety of datasets on cloud platforms including imagery, SAR data, and point clouds. Besides simple static catalog descriptions, STAC-API presents a web service to query items (e.g. images) of catalogs by space, time, and other properties. In R, the **rstac** package [@simoes_rstac_2021] allows to connect to STAC-API endpoints and search for items. In the example below, we request all images from the [Sentinel-2 Cloud-Optimized GeoTIFF (COG) dataset on Amazon Web Services](https://registry.opendata.aws/sentinel-2-l2a-cogs) that intersect with a predefined area and time of interest. The result contains all found images and their metadata and URLs pointing to actual files on AWS. 
+
+```{r 09-stac-example, eval = FALSE}
+library(rstac)
+
+# Connect to the STAC-API endpoint for Sentinel-2 data
+# and search for images intersecting our AOI
+s = stac("https://earth-search.aws.element84.com/v0")
+items = s |>
+  stac_search(collections = "sentinel-s2-l2a-cogs",
+              bbox = c(7.1, 51.8, 7.2, 52.8), 
+              datetime = "2020-01-01/2020-12-31") |>
+  post_request() |> items_fetch()
+```
+
+
+Cloud storage differs from local hard disks and traditional image file formats do not perform well in cloud-based geoprocessing. Broadly speaking, the cloud-optimized GeoTIFF format is a specific type of GeoTIFF that makes it possible to efficiently read only parts of an image from cloud storage. As a result, reading rectangular subsets of an image or reading images at lower resolution becomes much more efficient. As an R user, you don't have to install anything to work with COGs because  [GDAL](https://gdal.org) (and any package using it) can already work with COGs. However, keep in mind that the availability of COGs is a big plus while browsing through catalogs of data providers.
+
+For larger areas of interest, requested images are still relatively difficult to work with: they may use different map projections, may spatially overlap, and the spatial resolution often depends on the spectral band. The **gdalcubes** package [@appel_gdalcubes_2019] can be used to abstract from individual images and to create and process image collections as four-dimensional data cubes.
+
+The example below shows a minimal example to create a lower resolution (250m) maximum NDVI composite from the Sentinel-2 images returned by the previous STAC-API search.
+
+```{r 09-gdalcubes-example, eval = FALSE}
+library(gdalcubes)
+
+# Create an image collection object from STAC items
+collection = stac_image_collection(items$features, property_filter = function(x) {x[["eo:cloud_cover"]] < 10})
```

Comment:
Can you shortly explain the `x[["eo:cloud_cover"]] < 10` code?

Reply:
The comment above now states "Filter images from STAC response by cloud cover and create an image collection object". Is this understandable or do you think I should mention some details how the function is applied in the text?

Reply:
This is the only line of code in your sections that I do not understand (I do not know where did `"eo:cloud_cover"` came from...). It would be great if you could add a short explanation in the text.

Reply:
Thanks, I've added a short explanation ([see here](https://github.com/Robinlovelace/geocompr/pull/784/commits/2a203103209a02798cf0e90ff5f68ca9156aa18e#diff-56a79132069e015d3286d9bf8fa3ec5548a1d70c427f15f8a7f42b65ebdc9fcbR964-R966)) after the code chunk. Does this help to understand the function?